core/src/api_keys.rs

```diff
@@ -0,0 +1,80 @@
+use anyhow::{anyhow, Result};
+use axum::http::Request;
+use axum::middleware::Next;
+use axum::response::Response;
+use http::StatusCode;
+use serde::Deserialize;
+use std::{collections::HashMap, env, sync::Arc};
+use tokio::{fs, sync::OnceCell};
+use tracing::{error, info};
+
+// TODO(@fontanierh): Switch this to true once we have API keys configured.
+static ENFORCE_API_KEY_REQUIRED: bool = false;
+
+type ApiKeyMap = Arc<HashMap<String, Vec<String>>>;
+static API_KEYS: OnceCell<ApiKeyMap> = OnceCell::const_new();
+
+#[derive(Deserialize, Clone)]
+struct ApiKeyEntry {
+    client_name: String,
+    api_key: String,
+}
+
+async fn init_api_keys() -> Result<ApiKeyMap> {
```

Comment:
No because it is async. That's why a `OnceCell` is perfect here

Reply:
still it's a new import that could be avoided :/

Reply:
there is a block read_to_string in std which would be called only once here which is fine IMHO and would largely simplify the code?

Reply:
> largely simplify the code

It would save maximum 2 lines of code here

> still it's a new import that could be avoided :/

It would be the exact same number of imports because we'd need a lock instead

Reply:
And reading the file synchronously at a random time seems objectively worse than simply doing the right thing and using tokio no ?

Reply:
same imports # but one removed dependency. Everything would go in the lazy_static so you would not need your getter/setter and init function

Reply:
I agree that the sync read is not ideal though.

Reply:
> same imports # but one removed dependency.

Also same number of dependencie. OnceCell is coming from Tokio, which we already have